src/submission/validation-clinical/therapy.ts

```diff
@@ -17,42 +17,234 @@
+      treatment_type,
+    },
+  } = treatment;
+
+  const {
+    submitter_donor_id: submittedTherapyDonorId,
+    radiation_boost,
+    reference_radiation_treatment_id,
+  } = therapyRecord;
+
+  // Compare across other Submissions
+  let submittedTreatments: DeepReadonly<ClinicalInfo>[] = [];
+
+  for (const donorId in submittedRecords) {
+    const submittedTreatmentRecords = [...submittedRecords[donorId].treatment] as ClinicalInfo[];
+    if (submittedTreatmentRecords && submittedTreatmentRecords.length)
+      submittedTreatments = [...submittedTreatments, ...submittedTreatmentRecords];
+  }
+
+  const query: ClinicalQuery = {
+    programShortName: programId,
+    entityTypes: [ClinicalEntitySchemaNames.TREATMENT, ClinicalEntitySchemaNames.RADIATION],
+    page: 0,
+    sort: 'donorId',
+    donorIds: [],
+    submitterDonorIds: [],
+  };
+
+  // Compare across all Treatments
+  const { donors } =
+    (await donorDao.findByPaginatedProgramId(programId, query)) || emptyDonorResult;
+
+  const storedTreatments = donors.length
+    ? donors
+        .map(donor => donor.treatments)
+        .filter(treatments => Boolean(treatments))
+        .flat()
+    : [];
+
+  let errors: SubmissionValidationError[] = [];
+
+  if (typeof radiation_boost === 'string' && radiation_boost.toLowerCase() === 'yes') {
+    // Reference Radiation ID Matches a Submitted/Previous Submitter Treatment ID
+    const submittedTreatmentIdMatch = submitter_treatment_id === reference_radiation_treatment_id;
+
+    const submissionTreatmentIdMatch = submittedTreatments.find(
+      treatmentRecord =>
+        treatmentRecord.submitter_treatment_id === reference_radiation_treatment_id,
+    );
+
+    const storedTreatmentIdMatch = storedTreatments.find(treatmentRecord => {
+      const clinicalInfo = treatmentRecord?.clinicalInfo;
+      return clinicalInfo?.submitter_treatment_id === reference_radiation_treatment_id;
+    });
+
+    const treatmentIdMatch =
+      submittedTreatmentIdMatch || submissionTreatmentIdMatch || storedTreatmentIdMatch;
+
+    if (!treatmentIdMatch) {
+      errors = [
+        ...errors,
+        utils.buildSubmissionError(
+          therapyRecord,
+          DataValidationErrors.INVALID_REFERENCE_RADIATION_DONOR_ID,
+          TreatmentFieldsEnum.submitter_treatment_id,
+          {
+            [TreatmentFieldsEnum.treatment_type]: treatment_type,
+            reference_radiation_treatment_id,
+            therapyType: ClinicalEntitySchemaNames.RADIATION,
+          },
+        ),
+      ];
+    }
+
+    // Therapy + Treatment are associated with Radiation
+    const currentTreatmentIsRadiation = treatment_type && treatment_type === 'radiation';
+
+    const submittedTreatmentIsRadiation =
+      submissionTreatmentIdMatch && submissionTreatmentIdMatch.treatment_type === 'radiation';
+
+    const storedTreatmentIsRadiation =
+      storedTreatmentIdMatch && storedTreatmentIdMatch.clinicalInfo?.treatment_type === 'radiation';
```

Comment:
may as well just stick with optional chaining for the full block rather than half old school object check

Reply:
Updated